rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -0,0 +1,295 @@
+            )
+            return
+
+        examples = nlu_item.get(KEY_REGEX_EXAMPLES, "")
+        if not examples:
+            raise_warning(
+                f"Issue found while processing '{self.filename}': "
+                f"'{KEY_REGEX}: {regex_name}' doesn't have any examples. "
+                "It will be skipped.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        if not isinstance(examples, str):
+            raise_warning(
+                f"Unexpected block found in '{self.filename}':\n"
+                f"{examples}\n"
+                f"This block will be skipped.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        for example in examples.splitlines():
+            self.regex_features.append({"name": regex_name, "pattern": example})
+
+    def _parse_lookup(self, nlu_item: Dict[Text, Any]):
+        import rasa.nlu.training_data.lookup_tables_parser as lookup_tables_parser
+
+        lookup_item_name = nlu_item[KEY_LOOKUP]
+        if not lookup_item_name:
+            raise_warning(
+                f"Issue found while processing '{self.filename}': "
+                f"The lookup item has an empty name. "
+                f"Lookup items should have a name defined under the '{KEY_LOOKUP}'. "
+                "It will be skipped.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        examples = nlu_item.get(KEY_LOOKUP_EXAMPLES, "")
+        if not examples:
+            raise_warning(
+                f"Issue found while processing '{self.filename}':"
+                f"'{KEY_LOOKUP}: {lookup_item_name}' doesn't have any examples. "
+                "It will be skipped.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        if not isinstance(examples, str):
+            raise_warning(
+                f"Unexpected block found in '{self.filename}':\n"
+                f"{examples}\n"
+                f"This block will be skipped.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        for example in examples.splitlines():
+            lookup_tables_parser.add_item_to_lookup_tables(
+                lookup_item_name, example, self.lookup_tables
+            )
+
+    @staticmethod
+    def is_yaml_nlu_file(filename: Text) -> bool:
+        """Checks if the specified file possibly contains NLU training data in YAML.
+
+        Args:
+            filename: name of the file to check.
+
+        Returns:
+            `True` if the `filename` is possibly a valid YAML MLU file,
+            `False` otherwise.
+        """
+        if not Path(filename).suffix in YAML_FILE_EXTENSIONS:
+            return False
+        try:
+            content = io_utils.read_yaml_file(filename)
+            if KEY_NLU in content:
+                return True
+        except (YAMLError, Warning) as e:
```

Comment:
does the YAML lib throw the `Warning` as an exception?

Reply:
I think yes. I don't remember the particular example, but that happened during testing.